main.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/DMarby/picsum-photos/api"
+	fileDatabase "github.com/DMarby/picsum-photos/database/file"
+	"github.com/DMarby/picsum-photos/health"
+	vipsProcessor "github.com/DMarby/picsum-photos/image/vips"
+	"github.com/DMarby/picsum-photos/logger"
+	fileStorage "github.com/DMarby/picsum-photos/storage/file"
+	"go.uber.org/zap"
+)
+
+func waitForInterrupt(ctx context.Context) error {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+	select {
+	case sig := <-c:
+		return fmt.Errorf("received signal %s", sig)
+	case <-ctx.Done():
+		return errors.New("canceled")
+	}
+}
+
+const readTimeout = 5 * time.Second
+const writeTimeout = 30 * time.Second
+const maxImageSize = 5000 // The max allowed image width/height to be requested
+
+func main() {
+	// Set up commandline flags
+	listen := flag.String("listen", ":8080", "listen address")
+	loglevel := zap.LevelFlag("level", zap.InfoLevel, "log level (default \"info\") (debug, info, warn, error, dpanic, panic, fatal)")
```

Comment:
Maybe have this settable via env var as well for docker

Reply:
👍